pkg/wireguard/agent/agent.go

```diff
@@ -317,22 +317,26 @@ func (a *Agent) RestoreFinished(cm *clustermesh.ClusterMesh) error {
 
 	for _, p := range dev.Peers {
 		if pc, ok := pubKeyToPeerConfig[p.PublicKey]; ok {
+			var hasObsolete bool
 			for _, ip := range p.AllowedIPs {
 				if !pc.hasAllowedIP(ip) {
+					hasObsolete = true
 					pc.queueAllowedIPsRemove(ip)
 				}
 			}
 
-			log.WithFields(logrus.Fields{
-				logfields.Endpoint: pc.endpoint,
-				logfields.PubKey:   pc.pubKey,
-			}).Info("Removing obsolete AllowedIPs from WireGuard peer")
-			if err := a.updatePeerByConfig(pc); err != nil {
-				log.WithError(err).WithFields(logrus.Fields{
+			if hasObsolete {
```

Comment:
Is this necessary to reduce syscalls? The updated version of `updatePeerByConfig` already skips any syscalls if no IP removals are queued, right?

Reply:
Yes, you're right. We would still log `Removing obsolete ips...` even with no IP to be removed though, but this can be useful anyway to know that the restoration logic is running.
Removed this condition.